core/src/main/java/org/elasticsearch/env/Environment.java

```diff
@@ -126,6 +128,11 @@ public Environment(Settings settings) {
             dataFiles = new Path[]{homeFile.resolve("data")};
             dataWithClusterFiles = new Path[]{homeFile.resolve("data").resolve(ClusterName.clusterNameFromSettings(settings).value())};
         }
+        if (settings.get("path.shared_data") != null) {
+            sharedDataFile = PathUtils.get(cleanPath(settings.get("path.shared_data")));
+        } else {
+            sharedDataFile = homeFile.resolve("data");
```

Comment:
Ahh, part of the reason is that `path.data` can be an array, and so I didn't want to just randomly pick one of the paths